s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -238,6 +239,41 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
       .mapConcat(r => Stream.continually(r))
       .zip(Source.fromIterator(() => Iterator.from(1)))
 
+  //todo: this seems like a general purpose 'gadget', is there a better place for this kind of beasts?
+  private def conditionalAppendFl: Flow[ByteString, ByteString, NotUsed.type] =
```

Comment:
doesn't `Source.orElse` do this already?

Reply:
don't know, will check